ci/3-envs.yml

```diff
@@ -0,0 +1,427 @@
+        input_mapping: { repo: pre-production, version: rc-version }
+
+  - name: e2e-test-on-pre-production
+    serial: true
+    plan:
+      - aggregate:
+        - get: pre-production
+          resource: repo-pre-production
+          trigger: true
+          passed: [ ship-pre-production ]
+      - task: e2e-test
+        file: pre-production/ci/tasks/e2e.yml
+        input_mapping: { repo: pre-production }
+
+
+  - name: merge-pre-production-to-production
+    serial: true
+    plan:
+      - aggregate:
+        - get: pre-production
+          resource: repo-pre-production
+          passed: [ e2e-test-on-pre-production ]
+          trigger: true
+        - get: production
+          resource: repo-production
+      - task: merge-pre-production-to-production
+        file: pre-production/ci/tasks/merge-branch.yml
+        input_mapping: { from: pre-production, to: production }
+        output_mapping: { out: next-production }
+        params:
+          GIT_EMAIL: {{git-email}}
+          GIT_NAME: {{git-name}}
+          NO_FF: true
+      - put: production
+        resource: repo-production
+        params:
+          repository: next-production
+
+  - name: test-production
+    serial: true
+    plan:
+      - get: production
+        resource: repo-production
+        # comment out next line for support hotfix on production branch
+        passed: [ merge-pre-production-to-production ]
+        trigger: true
+      - task: unit-on-production
+        file: production/ci/tasks/unit.yml
+        input_mapping: { repo: production }
+
+  - name: version-production
+    serial: true
+    plan:
+      - aggregate:
+        - get: production
+          resource: repo-production
+          passed: [ test-production ]
+          trigger: true
+        - get: rc-version
+          params: { bump: final }
+      - put: final-version
+        params: { file: rc-version/version }
+
+  - name: tag-production
+    serial: true
+    plan:
+      - aggregate:
+        - get: production
+          resource: repo-production
+          passed: [ version-production ]
+          trigger: true
+        - get: final-version
+          passed: [ version-production ]
+      - put: production
+        resource: repo-production
+        params:
+          repository: production
+          tag: final-version/version
+
+  - name: ship-production
+    serial: true
+    plan:
+      - get: production
+        resource: repo-production
+        passed: [ tag-production ]
+        #Uncomment this line for CD
+        #trigger: true
+      - get: final-version
+        passed: [ tag-production ]
+      - task: ship-production
+        file: production/ci/tasks/shipit.yml
+        input_mapping: { repo: production, version: final-version }
+
+
+# semver control
+  - name: patch
+    serial: true
+    serial_groups: [ update-version ]
+    plan:
+      - get: dev-version
+      - put: dev-version
+        params: { bump: patch, pre: dev }
```

Comment:
manual bump jobs initializes `dev-version`.  So the same problem will occur with https://github.com/everpeace/concourse-gitlab-flow/pull/3/files#r77526922